src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -92,13 +103,21 @@ private static void RemoveAdditiveSpans(ArrayBuilder<ClassifiedSpan> spans)
 
             // The classification service will only produce classifications for
             // things it knows about.  i.e. there will be gaps in what it produces.
-            // Fill in those gaps so we have *all* parts of the span 
+            // By default, fill in those gaps so we have *all* parts of the span 
             // classified properly.
             using var _1 = ArrayBuilder<ClassifiedSpan>.GetInstance(out var filledInSyntaxSpans);
             using var _2 = ArrayBuilder<ClassifiedSpan>.GetInstance(out var filledInSemanticSpans);
 
-            FillInClassifiedSpanGaps(widenedSpan.Start, syntaxSpans, filledInSyntaxSpans);
-            FillInClassifiedSpanGaps(widenedSpan.Start, semanticSpans, filledInSemanticSpans);
+            if (fillInClassifiedSpanGaps)
+            {
+                FillInClassifiedSpanGaps(widenedSpan.Start, syntaxSpans, filledInSyntaxSpans);
+                FillInClassifiedSpanGaps(widenedSpan.Start, semanticSpans, filledInSemanticSpans);
+            }
+            else
+            {
+                filledInSyntaxSpans = syntaxSpans;
+                filledInSemanticSpans = semanticSpans;
```

Comment:
i don't love this.  primarily because you're overiwring a variable that feels like it is controlled by the using block.  Technically, this works... it's just oogy :)

Reply:
Gotcha. I slightly refactored this portion in my latest push just now, to me it looks a little cleaner but let me know if you still have any concerns.

Reply:
Ok, I actually had to revert my refactoring - guess the logic wasn't equivalent 😦 lmk if you have any other suggestions